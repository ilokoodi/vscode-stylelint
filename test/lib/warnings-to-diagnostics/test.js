'use strict';

const { lint } = require('stylelint');

const fn = require('../../../src/warnings-to-diagnostics');

describe('stylelintWarningToVscodeDiagnostic()', () => {
	test('should convert a stylelint warning into a VS Code diagnostic and consider severity level', async () => {
		expect.assertions(2);
		const {
			results: [{ warnings }],
		} = await lint({
			code: `a {
				color: #AAA;
				border-color: #bbbbbb;
			}`,
			config: {
				rules: {
					'color-hex-case': ['lower'],
					'color-hex-length': ['short', { severity: 'warning' }],
				},
			},
		});

		expect(fn(warnings[0])).toMatchSnapshot();
		expect(fn(warnings[1])).toMatchSnapshot();
	});
});
