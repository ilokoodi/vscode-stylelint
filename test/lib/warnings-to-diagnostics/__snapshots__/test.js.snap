// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stylelintWarningToVscodeDiagnostic() should convert a stylelint warning into a VS Code diagnostic and consider severity level 1`] = `
Object {
  "code": "color-hex-case",
  "message": "Expected \\"#AAA\\" to be \\"#aaa\\" (color-hex-case)",
  "range": Object {
    "end": Object {
      "character": 11,
      "line": 1,
    },
    "start": Object {
      "character": 11,
      "line": 1,
    },
  },
  "severity": 1,
  "source": "stylelint",
}
`;

exports[`stylelintWarningToVscodeDiagnostic() should convert a stylelint warning into a VS Code diagnostic and consider severity level 2`] = `
Object {
  "code": "color-hex-length",
  "message": "Expected \\"#bbbbbb\\" to be \\"#bbb\\" (color-hex-length)",
  "range": Object {
    "end": Object {
      "character": 18,
      "line": 2,
    },
    "start": Object {
      "character": 18,
      "line": 2,
    },
  },
  "severity": 2,
  "source": "stylelint",
}
`;
